Neos.Utility.Schema/Classes/TYPO3/Flow/Utility/SchemaValidator.php

```diff
@@ -624,4 +625,45 @@ protected function isDictionary(array $phpArray)
     {
         return array_keys($phpArray) !== range(0, count($phpArray) - 1);
     }
+
+    /**
+     * Generate a short description for the given schema-type to be used in error-messages
+     *
+     * @param mixed $type
+     * @return string
+     */
+    protected function getSchemaTypeInfo($type) {
+        if (is_scalar($type)) {
+            return (string)$type;
+        } else if (is_array($type)) {
+            if ($this->isSchema($type) && array_key_exists('type', $type)) {
+                return $this->getSchemaTypeInfo($type['type']);
+            } else if ($this->isNumericallyIndexedArray($type)) {
+                $subTypeInfos = [];
+                foreach ($type as $subType) {
+                    $subTypeInfos[] = $this->getSchemaTypeInfo($subType);
+                }
+                return '[' . implode(', ', $subTypeInfos) . ']';
+            } else {
+                return 'dictionary';
+            }
+        }
+        return gettype($type);
+    }
+
+    /**
+     * Generate a short description for the given value to be used in error-messages
+     *
+     * @param mixed $value
+     * @return string
+     */
+    protected function getValueInfo($value) {
+        if (is_scalar($value)) {
+            return (string)$value;
```

Comment:
I recommend ot always use `\TYPO3\Flow\Utility\TypeHandling::getTypeForValue` to get a type.

Also `(string)1`, `(string)true` and `(string)'1'`will have identitical return values, not sure if that is useful.